frappe/core/doctype/role/role.py

```diff
@@ -87,6 +88,89 @@ def update_user_type_on_change(self):
 				user.save()
 
 
+@frappe.whitelist()
+def duplicate_role(source_name, new_role_name=None, target_role=None):
```

Comment:
`target_role` is not sent from the frontend js. So, no point in having any logic around that. Please remove this param.